### ISSUE TITLE
Metabase seems to ignore the line breaks in multi-line text that it fetches from underlying DB

### ISSUE THREAD

**Issue by AshisRaj** [metabase/metabase#6178]
- Browser and the version:  Chrome 61.0.3163.100 (Official Build) (64-bit)
- Operating system: Windows 7 Professional SP1 (64-bit)
- Database: Vertica
- Metabase version: 0.25.2

# Abstract
When Metabase extracts out the multi-line text from DB, the line breaks seem to be ignored on UI. 

## Problem Statement
We have been publishing out test result's diff outputs into DB. But the diff content (being pulled by Metabase on UI) is very hard to read and understand as the line breaks/new lines are not being preserved in the data that is being displayed by the column(s) on the UI. Also because of this problem, the page stretches too much for user to keep scrolling to the right side.

Currently, there seem to be no way to workaround the problem we are facing.

## Suggested Solution
It would be better if there is something like a 'link' to the diff content provided there in the column(s) and those links take us to a popup box/page where we could only see the diff content with line breaks/new lines preserved.

![diff_content_without_linebreak](https://user-images.githubusercontent.com/32839577/31612290-e467bfe8-b29c-11e7-8c91-4e0a7afab00e.png)
![diff_content_without_linebreak_2](https://user-images.githubusercontent.com/32839577/31612293-e4af75ae-b29c-11e7-9f5f-57e5345e57d4.png)

**Comment by montanaflynn** [comment]
Yes please! I'd love to be able to select long form content, the current table implementation makes that harder than it should be. Glad the inspect element works though!

**Comment by ccemeraldeyes** [comment]
Also happens when attempting to insert line breaks in string_agg functions

**Comment by ipeevski** [comment]
As an alternative to the proposed solution, add a column visualisation option to show text over multiple lines or not. If ticked, it will expand the entire rows to be as tall as they need to be to respect new line characters within the selected columns

**Comment by ashleyhindle** [comment]
Adding column formatting of 'Multiline text' here would be perfect for me at least 👌 

<img width="341" alt="Screenshot 2022-03-28 at 21 11 48" src="https://user-images.githubusercontent.com/454975/160479088-8c8366cf-2629-47fd-90f1-2939d9fa95bb.png">

**Comment by skjbulcher** [comment]
I would also like the ability to display HTML or Markdown. Comments are important for some types of reporting, and we use various formats.

**Comment by dexter-atob** [comment]
I'm having trouble with query analyze because of this. Looking forward to this feature.

**Comment by pedro** [comment]
Adding a veery rough pass of Multiline display in #32200. I'm totally unfamiliar with the code here, hoping someone can give some pointers to get that to a mergeable state!

**Comment by erfanium** [comment]
I understand the reasoning behind #32200 and why it get closed by maintainers.

I suggest DBeaver idea. when you hover on a text field, it will show a tooltip, containing all the text data. this way, we don't need to change current table design.


![image](https://github.com/metabase/metabase/assets/47688578/97fa167c-6735-48f4-a55e-6fd5e0f33d32)

**Comment by bpartridge** [comment]
As a local override if you need this for your Item Detail modals, you can use a browser extension like [Stylus](https://add0n.com/stylus.html) and add the following CSS to metabaseapp.com or your server URL:

```
.ObjectDetail [colspan="2"] {
  white-space: pre; /* or break-spaces */
}
```

Then, if you just make that column `jsonb_pretty(my_jsonb_column)` in a SQL query, you get everything you need!

<img width="736" alt="Screenshot 2023-12-21 at 10 00 49 AM" src="https://github.com/metabase/metabase/assets/20748/6703a557-b934-461f-ac37-a060c5e7a938">

I'd love if there was an option for Text formatted columns to preserve newlines in the Detail visualization, and when they're rendered in the frontend, add white-space: pre or break-spaces there. I imagine that would require an extension to the data model for Text similar to the options for Links though.

**Comment by fjcero** [comment]
Any alternatives? Seems the most simple way is to fork the project and fix it?

**Comment by ipeevski** [comment]
In the latest version, there is an option to view the expanded record (icon at the beginning of the row). That helps in some cases

**Comment by icaropires** [comment]
> I'm having trouble an `EXPLAIN ANALYZE` query because of this. Looking forward to this feature.

If anyone else is also having trouble with `explain analyze`, one alternative is to download the output as csv. At least on version v0.49.7, the file has the formatting preserved

**Comment by brunobergher** [comment]
Highly related to #6826, which is hopefully coming soon.

**Comment by skjbulcher** [comment]
> Highly related to [#6826](https://github.com/metabase/metabase/issues/6826), which is hopefully coming soon.

There has been no significant activity on that issue for years. I wouldn't hold my breath.

Aesthetics (the main reason I've seen for not working on this functionality) seems like a weird excuse not to work on this functionality, since it must be specifically chosen by the user. I.e. the user has to deliberately choose this option to make the table to look a certain way.

**Comment by brunobergher** [comment]
There's a team working on it right now, @skjbulcher.

**Comment by skjbulcher** [comment]
I am very happy to be proven wrong. Thanks!

**Comment by dagi3d** [comment]
> There's a team working on it right now, [@skjbulcher](https://github.com/skjbulcher).

hi! what's the current status? I was using `STRING_AGG` in Postgres with a `\n` char hoping it would show multiple lines but it is printed right as "\n".
also tried "<br />" just in case but it did not work either.

**Comment by paoliniluis** [comment]
@dagi3d https://github.com/metabase/metabase/issues/6178#issuecomment-2573539348

**Comment by cdeweyx** [comment]
> > There's a team working on it right now, [@skjbulcher](https://github.com/skjbulcher).
> 
> hi! what's the current status? I was using `STRING_AGG` in Postgres with a `\n` char hoping it would show multiple lines but it is printed right as "\n". also tried "" just in case but it did not work either.

We added the notion of variable row height in v54 i.e. text wrapping so now we have a clear path towards allowing this. Our team plans to add this in the next few releases.

**Comment by moseleyi** [comment]
Respecting line breaks is needed for text fields!

**Comment by dominikj-cf** [comment]
This would really be amazing to have.

We're doing some bigger JSON aggregations/mappings and newlines are rather important for formatting.

Currently using an ugly hack of appending many spaces to each line to force a line break in the visualization output.
